controllers/storagecluster/server.go

```diff
@@ -0,0 +1,179 @@
+package storagecluster
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/multierr"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v1"
+)
+
+var (
+	ServerImage = "quay.io/sp1098/ocs-operator:consumer-api"
```

Comment:
Question to the maintainers: How should I feed this image name in the ocs-operator?

Reply:
I was under the impressions that we have decided to not have a separate image and that the image will be the ocs operator image (running from a different entry point)

Reply:
That is correct, But we still need to either feed the image or fetch it on runtime from ocs-operator deployment.

Reply:
Why cant we just read the image name from the ocs-operator CSV?

Reply:
> We can do that but for that we either need a CSV name or we need to filter it out with ocs-operator name where ocs-operator deployment name will be always the same.

We already do exact that in the deployer, you can take a peak here:
https://github.com/red-hat-storage/ocs-osd-deployer/blob/fbc8169d7b7a81147309788813b07c2a88031aa1/controllers/managedocs_controller.go#L1389-L1399

> As far as I know, metrics exporter deployment is a part of the CSV which gets installed on the operator install not on storageCluster installation. So that's a different usecase.

Ok

Reply:
So you would like to filter with the `ocs-operator` prefix. ok I can do that

Reply:
This is just one way to do it. Another option is to specify it in an env var as part of the CSV. That way it will be similar to all the other stuff that is already mentioned in the CSV.

Reply:
Could you just use downward API field references? https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/#the-downward-api e.g.:

```
      env:
        - name: IMAGE
          valueFrom:
            fieldRef:
              fieldPath: spec.containers[0].image
```

Reply:
@jarrpa @ issues here:
1. Referencing any container by an index is pruned to future issues due to container list location changes in future updates. 
2.  I don't think that the downward API support arbitrary spec fields, if I remember correctly there is a list of what can and can't be refernced to. But I might be wrong here.

Reply:
Looks like it. Okay, set the image as an annotation: https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/#capabilities-of-the-downward-api

Reply:
@jarrpa But why? what is the diff between setting it directly in the ENV var or setting an annotation and then referencing it in the env var? You still need to have it manually set up somewhere!

Reply:
In general, hard-coding container images in code feels wrong. Unless there's a good reason to do it, the default should be to avoid it at all costs.

Here specifically, however, if we take the definition out of the go code and into the manifests that makes it easier for us to integrate it into our make/kustomize workflow so we don't have to worry about keeping track of it separately from what we're already doing to generate the bundle. It also gives us more flexibility as to where it can live, though off-hand if makes more sense (to me) to put the annotation on the Pod itself. Since the spec is generated by automation form the same value, I'm not to worried about writing it to multiple places.

Reply:
Ah-ha... just talked to Ohad and I realized I was thinking about the wrong thing. Still, literally because we have the whole kustomize automation process, we can (and probably should) just inject the container image name directly into the Pod's environment variable. Let's do that!

Reply:
done